client/allocrunner/taskrunner/secrets/plugin_provider.go

```diff
@@ -38,26 +34,17 @@ func NewExternalPluginProvider(plugin commonplugins.SecretsPlugin, name string,
 	}
 }
 
-func (p *ExternalPluginProvider) Fetch(ctx context.Context) error {
+func (p *ExternalPluginProvider) Fetch(ctx context.Context) (map[string]string, error) {
 	resp, err := p.plugin.Fetch(ctx, p.path)
 	if err != nil {
-		return fmt.Errorf("failed to fetch secret from plugin %s: %w", p.name, err)
+		return nil, fmt.Errorf("failed to fetch secret from plugin %s: %w", p.name, err)
 	}
 	if resp.Error != nil {
-		return fmt.Errorf("error returned from secret plugin %s: %s", p.name, *resp.Error)
-	}
-
-	p.response = resp
-	return nil
-}
-
-func (p *ExternalPluginProvider) Parse() (map[string]string, error) {
-	if p.response == nil {
-		return nil, errors.New("no plugin response for provider to parse")
+		return nil, fmt.Errorf("error returned from secret plugin %s: %s", p.name, *resp.Error)
 	}
 
 	formatted := map[string]string{}
```

Comment:
We could possibly pre-allocate this map as we know the length of `resp.Result` but I don't believe it will make any difference in reality.